### PR TITLE
encryption restoration details

### PR DESCRIPTION
**Pull request by AnuM361**
Version(s): 4.3+

Issue:  More details about backup/restore of encrypted etcd

Link to docs preview:  https://docs.openshift.com/container-platform/4.11/security/encrypting-etcd.html

QE review:-  QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by EmilyM1** [comment]
@lahinson and @tmalove would you mind taking a look? 
This is to handle the case of encryption on restore for etcd.

File: modules/about-etcd-encryption.adoc [hunk 1]
```diff
@@ -22,4 +22,6 @@ When you enable etcd encryption, encryption keys are created. These keys are rot
 [NOTE]
 ====
 Keep in mind that etcd encryption only encrypts values, not keys. This means that resource types, namespaces, and object names are unencrypted.
+
+During etcd backup, if etcd encryption is enabled then static_kuberesources_<datetimestamp>.tar.gz: file also contains the encryption keys for the etcd snapshot and it is recommended to store this file separately from the etcd snapshot for security reasons. However, this file is required to restore a previous state of etcd from respective etcd snapshot.
```
Comment (tmalove) [inline modules/about-etcd-encryption.adoc:26]:
```suggestion
If etcd encryption is enabled during a backup, the static_kuberesources_<datetimestamp>.tar.gz: file contains the encryption keys for the etcd snapshot. You should store this file separately from the etcd snapshot for security reasons. However, this file is required to restore a previous state of etcd from the respective etcd snapshot.
```

**Comment by lahinson** [comment]
@EmilyM1 I added a couple of comments. Otherwise, LGTM!

**Review by tmalove — APPROVED** [review]
>

**Comment by asme30** [comment]
Thank you teams @lahinson  verified all the changes and it looks good

**Comment by lahinson** [comment]
Great, thanks @asme30! Can you incorporate the edits, or is that something that @tmalove and I need to help with?

**Comment by asme30** [comment]
@lahinson  and @tmalove Please guide me with this.

File: modules/about-etcd-encryption.adoc [hunk 2]
```diff
@@ -23,5 +23,5 @@ When you enable etcd encryption, encryption keys are created. These keys are rot
 ====
 Keep in mind that etcd encryption only encrypts values, not keys. This means that resource types, namespaces, and object names are unencrypted.
 
-During etcd backup, if etcd encryption is enabled then static_kuberesources_<datetimestamp>.tar.gz: file also contains the encryption keys for the etcd snapshot and it is recommended to store this file separately from the etcd snapshot for security reasons. However, this file is required to restore a previous state of etcd from respective etcd snapshot.
+If etcd encryption is enabled during a backup, the `__static_kuberesources_<datetimestamp>.tar.gz: file__` contains the encryption keys for the etcd snapshot. You should store this file separately from the etcd snapshot for security reasons. However, this file is required to restore a previous state of etcd from the respective etcd snapshot.
```
Comment (lahinson) [inline modules/about-etcd-encryption.adoc:26]: @asme30 To keep the text aligned with IBM and Red Hat style guidance, please update the second sentence from `You should store this file separately...` to `Store this file separately...`. After that change has been made, I can help you move this forward to be reviewed and merged by the OpenShift docs team. Thanks!

**Review by lahinson — CHANGES_REQUESTED** [review]
See my comment about line 26.

**Comment by asme30** [comment]
Hey @lahinson and @tmalove  Added recommended changes please review and let me know.

**Comment by lahinson** [comment]
@asme30 Looks good. Thanks!

@geliu2016 PTAL at this small change to the docs when you can.

File: modules/about-etcd-encryption.adoc [hunk 3]
```diff
@@ -22,4 +22,6 @@ When you enable etcd encryption, encryption keys are created. These keys are rot
 [NOTE]
 ====
 Keep in mind that etcd encryption only encrypts values, not keys. This means that resource types, namespaces, and object names are unencrypted.
```
[the remark this answers is not in the document]
Comment (gangwgr) [reply modules/about-etcd-encryption.adoc]: @geliu2016 I'm not sure, will check with @wangke19 and dev

[the remark this answers is not in the document]
Comment (gangwgr) [reply modules/about-etcd-encryption.adoc]: @lahinson Please find the details in slack https://coreos.slack.com/archives/CC3CZCQHM/p1670999032684989.
What other info you need?

**Comment by gangwgr** [comment]
@geliu2016 @lahinson Doc looks find to me.

**Comment by lahinson** [comment]
Thanks, @geliu2016 @gangwgr!

**Review by lahinson — APPROVED** [review]

**Comment by lahinson** [comment]
@asme30 When you can, please update the first comment in this PR to indicate that QE has approved this change.

**Comment by lahinson** [comment]
/label peer-review-needed

Hi peer review squad - This XS update to the etcd docs has been reviewed and approved by engineering and QE. PTAL. Thanks!

File: modules/about-etcd-encryption.adoc [hunk 4]
```diff
@@ -21,5 +21,7 @@ When you enable etcd encryption, encryption keys are created. These keys are rot
 
 [NOTE]
 ====
-Keep in mind that etcd encryption only encrypts values, not keys. This means that resource types, namespaces, and object names are unencrypted.
+Keep in mind that etcd encryption only encrypts values, not keys. Resource types, namespaces, and object names are unencrypted.
```
Comment (kelbrown20) [inline modules/about-etcd-encryption.adoc:24]: I don't think that the "Keep in mind" part is really necessary. This statement works on its own 
```suggestion
Etcd encryption only encrypts values, not keys. Resource types, namespaces, and object names are unencrypted.
```

File: modules/about-etcd-encryption.adoc [hunk 5]
```diff
@@ -21,5 +21,7 @@ When you enable etcd encryption, encryption keys are created. These keys are rot
...
+
+If etcd encryption is enabled during a backup, the `__static_kuberesources_<datetimestamp>.tar.gz: file__` contains the encryption keys for the etcd snapshot. Store this file separately from the etcd snapshot for security reasons. However, this file is required to restore a previous state of etcd from the respective etcd snapshot.
```
Comment (kelbrown20) [inline modules/about-etcd-encryption.adoc:26]: Up to you! 
```suggestion
If etcd encryption is enabled during a backup, the `__static_kuberesources_<datetimestamp>.tar.gz: file__` contains the encryption keys for the etcd snapshot. For security reasons, store this file separately from the etcd snapshot. However, this file is required to restore a previous state of etcd from the respective etcd snapshot.
```

**Review by kelbrown20 — COMMENTED** [review]
Hey Laura! Great work! Just some minor suggestions to consider, otherwise this LGTM!

**Comment by lahinson** [comment]
Thanks, @kelbrown20! Good suggestions. @asme30 Please see the two comments from Kelly and make the updates. After that, we can move this forward to be merged.

**Comment by lahinson** [comment]
LGTM!

File: modules/about-etcd-encryption.adoc [hunk 6]
```diff
@@ -21,5 +21,7 @@ When you enable etcd encryption, encryption keys are created. These keys are rot
 
 [NOTE]
 ====
-Keep in mind that etcd encryption only encrypts values, not keys. This means that resource types, namespaces, and object names are unencrypted.
+Etcd encryption only encrypts values, not keys. Resource types, namespaces, and object names are unencrypted.
+
+If etcd encryption is enabled during a backup, the `__static_kuberesources_<datetimestamp>.tar.gz: file__` contains the encryption keys for the etcd snapshot. For security reasons, store this file separately from the etcd snapshot. However, this file is required to restore a previous state of etcd from the respective etcd snapshot.
```
Comment (jboxman-rh) [inline modules/about-etcd-encryption.adoc:26]: @asme30 

I hate to dig this up again, but this still looks wrong. I've never seen a filename of the form: static_kuberesources_<datetimestamp>.tar.gz: file, so I'm not sure this is correct? Or is it: static_kuberesources_<datetimestamp>.tar.gz?

Comment (jboxman-rh) [inline modules/about-etcd-encryption.adoc:26]:
```suggestion
If etcd encryption is enabled during a backup, the `__static_kuberesources_<datetimestamp>.tar.gz__` file contains the encryption keys for the etcd snapshot. For security reasons, store this file separately from the etcd snapshot. However, this file is required to restore a previous state of etcd from the respective etcd snapshot.
```

**Review by jboxman-rh — COMMENTED** [review]
I have a question before merging, thanks!

**Comment by asme30** [comment]
@jboxman-rh made suggested changes. PTAL

**Pull request merged**